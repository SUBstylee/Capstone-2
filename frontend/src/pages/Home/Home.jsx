import { useEffect } from "react";
import {useDispatch,useSelector} from 'react-redux'
import { Col, Row } from "react-bootstrap";
import Product from "../../components/Product/Product";
import Message from "../../components/Message/Message";
import Loader from '../../components/Loader/Loader';
import { listProducts } from "../../actions/productActions";

const Home = () => {
  const dispatch=useDispatch();
  const productList=useSelector(state=>state.productList);
  const {loading,error,products}=productList;

  useEffect(() => {
    dispatch(listProducts())
  }, [dispatch]);

  return (
    <div className='Home'>
      <h1>Latest Products</h1>
      {loading?(
        <Loader/>
      ):error?(
        <Message variant='danger'>{error}</Message>
      ):(
        <Row>
          {products.map((product) => (
            <Col sm={12} md={6} lg={4} key={product._id}>
              <Product product={product} />
            </Col>
          ))}
        </Row>
      )}
    </div>
  );
};

export default Home;
