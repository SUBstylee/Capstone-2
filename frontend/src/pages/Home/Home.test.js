import { render } from '@testing-library/react';
import Home from './Home';
import store from '../../store';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

test("renders Home component without crashing", () => {
    const { getByText } = render(
        <Provider store={store}><BrowserRouter><Home /></BrowserRouter></Provider>
    );

    expect(getByText("Latest Products")).toBeInTheDocument();
});

test("matches snapshot", function () {
    const { asFragment } = render(
        <Provider store={store}><BrowserRouter><Home /></BrowserRouter></Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});