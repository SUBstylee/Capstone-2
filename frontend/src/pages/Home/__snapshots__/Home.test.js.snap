// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Home"
  >
    <h1>
      Latest Products
    </h1>
    <div
      class="Loader"
    >
      <span
        class="sr-only"
      >
        Loading...
      </span>
    </div>
  </div>
</DocumentFragment>
`;
